alf/metrics/metric.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2020 Horizon Robotics. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""A few metrics.
+Code adapted from https://github.com/tensorflow/agents/blob/master/tf_agents/metrics/tf_metric.py
+"""
+
+import alf
+import os
+
+from torch import nn
+
+
+class StepMetric(nn.Module):
```

Comment:
What's the reason for deriving from `nn.Module`? Is it because for saving checkpoints?

Reply:
Yes, I think only nn.Module can be checkpointed automatically?

Reply:
Can add tests for all these sub-classes that the state_dict() give you the correct things you want to save in checkpoint.

Reply:
Good point, done.